src/query/procedure/mg_procedure_impl.hpp

```diff
@@ -545,14 +545,38 @@ struct mgp_result {
   std::optional<utils::pmr::string> error_msg;
 };
 
+struct mgp_func_result {
+  explicit mgp_func_result(mgp_value *value, std::optional<utils::pmr::string> error_msg)
+      : value(value), error_msg(error_msg) {}
+
+  explicit mgp_func_result() {}
+  /// Return Magic function result with potential error
+  mgp_value *value;
```

Comment:
This is where I am stuck currently. I've chosen `*value` here because in the lambda which is executing the external function I am defining `mgp_function_result` on the stack, and this value gets deleted when returned. After returning the value, this fails: `MG_ASSERT(res.GetMemoryResource() == ctx_->memory);`

Reply:
Holding the `TypedValue` in the `std::optional` container now.